Monika After Story/game/script-greetings.rpy

```diff
@@ -4389,38 +4423,176 @@ label greeting_spacing_out:
     m 1hubsb "Ahaha~"
     m 1eua "I'm very happy to see you again. {w=0.2}{nw}"
     extend 3eua "What should we do today, [player]?"
-
     return
 
 init 5 python:
-    gmr.eardoor.append("monikaroom_greeting_ear_recursionerror")
+    ev_rules = dict()
+    ev_rules.update(
+        MASGreetingRule.create_rule(
+            skip_visual=True,
+            random_chance=20,
+            override_type=True
+        )
+    )
+    ev_rules.update(
+        MASTimedeltaRepeatRule.create_rule(
+            datetime.timedelta(days=3)
+        )
+    )
 
-label monikaroom_greeting_ear_recursionerror:
-    m "Hmm, now that looks good. Let's-{w=0.5}{nw}"
-    m "Wait, no. Gosh, how did I forget..."
-    m "This has to be called right here."
+    addEvent(
+        Event(
+            persistent.greeting_database,
+            eventlabel="greeting_after_bath",
+            conditional=(
+                "mas_getAbsenceLength() >= datetime.timedelta(hours=6) "
+                "and not mas_isSpecialDay()"
+            ),
+            unlocked=True,
+            rules=ev_rules,
+            aff_range=(mas_aff.LOVE, None)
+        ),
+        code="GRE"
+    )
+
+    del ev_rules
+
+# TODO: move this to change_clothes, also add a similar var for hair
+default persistent._mas_previous_clothes = mas_clothes_def.name
```

Comment:
the `MASMonika` class has a [`save_state` function](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/sprite-chart.rpy#L3532) that can be used to get a snapshot of Monika's current outfit. The snapshot can be saved as primitive values so it can be saved to persistent. 

You can reload an entire outfit using that snapshot (`load_state` function).

If you need to do a comparison between snapshot and the current outfit, you can use the `same_state` function on the class to compare states - however this checks everything. To check a specific piece you will have to manually select the data from the tuple - the `save_state` function explains the order of the data.

Reply:
Do you want to save the complete data in persistent (clothes/hair/acs)? Because otherwise it's easier to copy clothes from `monika_chr.clothes.name` than to get a tuple and then parse its items.

Reply:
save the complete data so in the future we don't have to migrate persistent data or add additional vars.

I will probably add extraction functions later so you can get specific pieces out of the tuple state.

Reply:
Should be better now.